api_app/serializers.py

```diff
@@ -592,3 +597,77 @@ class EnvelopeSerializer(rfs.Serializer):
         results = serializer_class(many=many)
 
     return EnvelopeSerializer
+
+
+class CustomConfigSerializer(rfs.ModelSerializer):
+    def validate_value(self, value):
+        try:
+            json.loads(value)
+        except json.JSONDecodeError:
+            raise ValidationError("Value is not JSON-compliant.")
+        return value
+
+    def validate(self, attrs):
+        super().validate(attrs)
+
+        # check if owner is admin of organization
+        if attrs.get("organization", None):
+            # check if the user is owner of the organization
+            membership = Membership.objects.filter(
+                user=attrs.get("owner"),
+                organization=attrs.get("organization"),
+                is_owner=True,
+            )
+            if not membership.exists():
+                raise ValidationError("User is not owner of the organization.")
+
+        config = None
+        category = None
+        if attrs["type"] == CustomConfig.Type.ANALYZER:
+            config = AnalyzerConfig
+            category = "Analyzer"
+        elif attrs["type"] == CustomConfig.Type.CONNECTOR:
+            config = ConnectorConfig
+            category = "Connector"
+
+        if attrs["name"] not in config.all():
+            raise ValidationError(f"{category} {attrs['name']} does not exist.")
+
+        if attrs["attribute"] not in config.all()[attrs["name"]].params:
+            raise ValidationError(
+                f"{category} {attrs['name']} does not "
+                f"have attribute {attrs['attribute']}."
+            )
+
+        # Check if the type of value is valid for the attribute.
+        # NOTE: json.loads is used as it allows us to store all
+        # valid value types into a StringField
+        if not isinstance(
+            json.loads(attrs["value"]),
+            type(config.all()[attrs["name"]].params[attrs["attribute"]].value),
+        ):
+            expected_type = type(
+                config.all()[attrs["name"]].params[attrs["attribute"]].value
+            )
+            raise ValidationError(
+                f"{category} {attrs['name']} attribute "
+                f"{attrs['attribute']} has wrong type "
+                f"{type(json.loads(attrs['value'])).__name__}. Expected: "
+                f"{expected_type.__name__}."
+            )
+
+        if self.instance is None:
+            try:
+                params = attrs.copy()
+                params.pop("value")
+                print("params:", params)
+                self.instance = CustomConfig.objects.get(**params)
+                logger.info(f"CustomConfig {self.instance} found. Will be updated.")
+            except CustomConfig.DoesNotExist:
+                # No matching CustomConfig was found. A new entry will be created.
+                pass
+        return attrs
+
+    class Meta:
+        model = CustomConfig
+        fields = rfs.ALL_FIELDS
```

Comment:
that's cool, could you just move it at the start of the class CustomConfigSerializer? In that way a reader would read this first so he can understand which fields are available in this serializer before reading the rest of the code